tests/test_PR114_rpad_operation.py

```diff
@@ -0,0 +1,87 @@
+
+import awkward1
+
+def test_rpad_empty_array():
+    empty = awkward1.layout.EmptyArray()
+
+    assert awkward1.tolist(empty) == []
+
+def test_rpad_indexed_array():
+    array = []
+
+def test_rpad_numpy_array():
+    array = awkward1.layout.NumpyArray(numpy.arange(2*3*5, dtype=numpy.int64).reshape(2, 3, 5))
+    assert awkward1.tolist(array) == [[[ 0,  1,  2,  3,  4], [ 5,  6,  7,  8,  9], [10, 11, 12, 13, 14]],
+                                      [[15, 16, 17, 18, 19], [20, 21, 22, 23, 24], [25, 26, 27, 28, 29]]]
+    print(awkward1.tolist(array.rpad(5, 0)))
+    assert awkward1.tolist(array.rpad(5, 0)) == [[[0, 1, 2, 3, 4], [5, 6, 7, 8, 9], [10, 11, 12, 13, 14]],
+                                                [[15, 16, 17, 18, 19], [20, 21, 22, 23, 24], [25, 26, 27, 28, 29]],
+                                                None, None, None]
+
+    assert awkward1.tolist(array.rpad(5, 1)) == [[[0, 1, 2, 3, 4], [5, 6, 7, 8, 9], [10, 11, 12, 13, 14], None, None],
+                                                [[15, 16, 17, 18, 19], [20, 21, 22, 23, 24], [25, 26, 27, 28, 29], None, None]]
+
+    assert awkward1.tolist(array.rpad(7, 2)) == [[[0, 1, 2, 3, 4, None, None], [5, 6, 7, 8, 9, None, None], [10, 11, 12, 13, 14, None, None]],
+                                                [[15, 16, 17, 18, 19, None, None], [20, 21, 22, 23, 24, None, None], [25, 26, 27, 28, 29, None, None]]]
+
+    assert awkward1.tolist(array.rpad(2, 2)) == [[[0, 1], [5, 6], [10, 11]], [[15, 16], [20, 21], [25, 26]]]
+
+def test_rpad_regular_array():
+    content = awkward1.layout.NumpyArray(numpy.array([2.1, 8.4, 7.4, 1.6, 2.2, 3.4, 6.2, 5.4, 1.5, 3.9, 3.8, 3.0, 8.5, 6.9, 4.3, 3.6, 6.7, 1.8, 3.2]))
+    index = awkward1.layout.Index64(numpy.array([13, 9, 13, 4, 8, 3, 15, -1, 16, 2, 8], dtype=numpy.int64))
+    indexedarray = awkward1.layout.IndexedOptionArray64(index, content)
+    array = awkward1.layout.RegularArray(indexedarray, 3)
+
+    assert awkward1.tolist(array.rpad(5, 0)) == [[6.9, 3.9, 6.9], [2.2, 1.5, 1.6], [3.6, None, 6.7], [None, None, None], [None, None, None]]
+    assert awkward1.tolist(array.rpad(4, 0)) == [[6.9, 3.9, 6.9], [2.2, 1.5, 1.6], [3.6, None, 6.7], [None, None, None]]
+    assert awkward1.tolist(array.rpad(3, 0)) == [[6.9, 3.9, 6.9], [2.2, 1.5, 1.6], [3.6, None, 6.7]]
+    assert awkward1.tolist(array.rpad(2, 0)) == [[6.9, 3.9, 6.9], [2.2, 1.5, 1.6]]
+    assert awkward1.tolist(array.rpad(1, 0)) == [[6.9, 3.9, 6.9]]
+    assert awkward1.tolist(array.rpad(5, 1)) == [[6.9, 3.9, 6.9, None, None], [2.2, 1.5, 1.6, None, None], [3.6, None, 6.7, None, None]]
+    assert awkward1.tolist(array.rpad(4, 1)) == [[6.9, 3.9, 6.9, None], [2.2, 1.5, 1.6, None], [3.6, None, 6.7, None]]
+    assert awkward1.tolist(array.rpad(3, 1)) == [[6.9, 3.9, 6.9], [2.2, 1.5, 1.6], [3.6, None, 6.7]]
+    assert awkward1.tolist(array.rpad(2, 1)) == [[6.9, 3.9], [2.2, 1.5], [3.6, None]]
+    assert awkward1.tolist(array.rpad(1, 1)) == [[6.9], [2.2], [3.6]]
```

Comment:
Okay, so this is always "pad and clip," the operation that always returns RegularArrays. There's also a "pad without clipping" that returns ListArrays, in which the length is ensured to be _at least_ a given value. The latter is used to make it possible to apply slices that assume there's a value "at index N" without also eliminating data beyond the interesting region.

Reply:
I'll rename `rpad` to `rpad_and_clip`

Reply:
done